SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -605,24 +606,54 @@ FRPCErrorInfo USpatialSender::SendRPC(const FPendingRPCParams& Params)
 {
 	SCOPE_CYCLE_COUNTER(STAT_SpatialSenderSendRPC);
 
+	bool bShouldDrop = false;
+
+	const float TimeDiff = (FDateTime::Now() - Params.Timestamp).GetTotalSeconds();
+	if (GetDefault<USpatialGDKSettings>()->QueuedOutgoingRPCWaitTime < TimeDiff)
+	{
+		bShouldDrop = true;
+	}
+
 	TWeakObjectPtr<UObject> TargetObjectWeakPtr = PackageMap->GetObjectFromUnrealObjectRef(Params.ObjectRef);
 	if (!TargetObjectWeakPtr.IsValid())
 	{
 		// Target object was destroyed before the RPC could be (re)sent
-		return FRPCErrorInfo{ nullptr, nullptr, NetDriver->IsServer(), ERPCQueueType::Send, ERPCResult::UnresolvedTargetObject };
+		return FRPCErrorInfo{ nullptr, nullptr, NetDriver->IsServer(), ERPCQueueType::Send, ERPCResult::UnresolvedTargetObject, bShouldDrop };
 	}
 	UObject* TargetObject = TargetObjectWeakPtr.Get();
 
 	const FClassInfo& ClassInfo = ClassInfoManager->GetOrCreateClassInfoByObject(TargetObject);
 	UFunction* Function = ClassInfo.RPCs[Params.Payload.Index];
 	if (Function == nullptr)
 	{
-		return FRPCErrorInfo{ TargetObject, nullptr, NetDriver->IsServer(), ERPCQueueType::Send, ERPCResult::MissingFunctionInfo };
+		return FRPCErrorInfo{ TargetObject, nullptr, NetDriver->IsServer(), ERPCQueueType::Send, ERPCResult::MissingFunctionInfo, bShouldDrop };
+	}
+
+	if (AActor* TargetActor = Cast<AActor>(TargetObject))
+	{
+		if (TargetActor->IsPendingKillPending())
+		{
+			bShouldDrop = true;
```

Comment:
Just return immediately

Reply:
Can do, I had this variable here for readability. Will pass `true` as one of the arguments on the next line instead.